lib/base-compiler.ts

```diff
@@ -3671,7 +3671,7 @@ but nothing was dumped. Possible causes are:
         if (this.compiler.supportsMarch) all.push([`-march=${c_value_placeholder}`]);
         if (this.compiler.supportsTargetIs) all.push([`--target=${c_value_placeholder}`]);
         if (this.compiler.supportsTarget) all.push(['--target', c_value_placeholder]);
-
+        if (this.compiler.supportsHyphenTarget) all.push(['-target', c_value_placeholder]);
```

Comment:
I think these last two might conflict and return 2x a target. Perhaps and Else would be in order?

Reply:
After reviewing the code again, I don't believe they will conflict. This PR only adds support for `this.compiler.supportsHyphenTarget` to the `ZigParser` class, which will not see `this.compiler.supportsTarget`. I don't believe there is a conflict here.

Reply:
But in general it would be surprising if a user added both, without understanding that much further on this would cause a problem. However, that already exists between `supportsTargetIs` and `supportsTarget`, so I think probably this is as ok as it was before (cc @partouf in case you disagree?). We can file an improvement request to update the whole supportsTarget thing, perhaps turning it into a enum type thing.